lib/DBD/MariaDB.pm

```diff
@@ -64,36 +64,30 @@ sub CLONE {
 
 sub parse_dsn {
     my ($class, $dsn) = @_;
+
     my $hash = {};
-    my($var, $val);
-    if (!defined($dsn)) {
-        return $hash;
-    }
     while (length($dsn)) {
-	if ($dsn =~ /([^:;]*\[.*]|[^:;]*)[:;](.*)/) {
-	    $val = $1;
-	    $dsn = $2;
-	    $val =~ s/\[|]//g; # Remove [] if present, the rest of the code prefers plain IPv6 addresses
+	my $part;
+	if ($dsn =~ /([^:;]*\[[^\]]*\]|[^:;]*)[:;](.*)/) {
+	    ($part, $dsn) = ($1, $2);
+	    $part =~ tr/\[\]//d; # Remove [] if present, the rest of the code prefers plain IPv6 addresses
 	} else {
-	    $val = $dsn;
-	    $dsn = '';
+	    ($part, $dsn) = ($dsn, '');
 	}
-	if ($val =~ /([^=]*)=(.*)/) {
-	    $var = $1;
-	    $val = $2;
-	    if ($var eq 'hostname'  ||  $var eq 'host') {
-		$hash->{'host'} = $val;
-	    } elsif ($var eq 'db'  ||  $var eq 'dbname') {
-		$hash->{'database'} = $val;
-	    } else {
-		$hash->{$var} = $val;
+	if ($part =~ /([^=]*)=(.*)/) {
+	    my ($var, $val) = ($1, $2);
+	    # These are legacy, do not change, and do not add
+	    if ( $var eq 'db' || $var eq 'dbname' ) {
```

Comment:
Please do not mix tabs and spaces as it just cause problems for readers who have different tabstop settings in editor. If you are rewriting whole function (like this one) then please use 4 spaces (not tabs) for indentation.

Reply:
I was previously using the GitHub editor, which used whatever indentation was already there.  My most recent changes have updated to be space-only indentations.